pkg/services/publicdashboards/api/api_test.go

```diff
@@ -49,6 +50,61 @@ type JsonErrResponse struct {
 	Error string `json:"error"`
 }
 
+func TestAPIGetAnnotations(t *testing.T) {
+	testCases := []struct {
+		Name                 string
+		ExpectedHttpResponse int
+		Annotations          []AnnotationEvent
+		ServiceError         error
+		From                 string
+		To                   string
+	}{
+		{
+			Name:                 "will return success when there is no error and to and from are provided",
+			ExpectedHttpResponse: http.StatusOK,
+			Annotations:          []AnnotationEvent{{Id: 1}},
+			ServiceError:         nil,
+			From:                 "123",
+			To:                   "123",
+		},
+		{
+			Name:                 "will return 500 when service returns an error",
+			ExpectedHttpResponse: http.StatusInternalServerError,
+			Annotations:          nil,
+			ServiceError:         errors.New("an error happened"),
+			From:                 "123",
+			To:                   "123",
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.Name, func(t *testing.T) {
+			cfg := setting.NewCfg()
+			cfg.RBACEnabled = false
+			service := publicdashboards.NewFakePublicDashboardService(t)
+			if test.ServiceError != nil {
+				service.On("GetAnnotations", mock.Anything, mock.Anything, mock.AnythingOfType("string")).
```

Comment:
We can remove the if statement here

`service.On("GetAnnotations", mock.Anything, mock.Anything, mock.AnythingOfType("string")).				Return(test.Annotations, test.ServiceError).Maybe()`

and then just make sure the values are set correctly in the struct

Reply:
nice